### PR TITLE
Reference TypeScript instead of JavaScript in documentation

### PR DESCRIPTION
**Pull request by kasperg**
#### Description

This should clarify the primary language of the project.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -278,7 +275,7 @@ appropriate.
 
 ```scss
 // ./src/apps/my-new-application/my-new-application.scss
-.dpl-warm {
+.warm {
```
Comment (hrmoller) [inline README.md:278]: Doesn't the removal of `dpl-` contradict with what's stated in [`code_guidelines.md` line 94](https://github.com/reload/dpl-react/blob/docs-typescript/docs/code_guidelines.md?plain=1#L94)?

**Review by hrmoller — COMMENTED** [review]
A few typos... I think:)

- Besides what I consider typos which has been commented on on the specific lines I have found a few places in the README where the term "atoms" is still present even though it seems as the usage of that term is discontinued.
  1. [On line 421](https://github.com/reload/dpl-react/blob/docs-typescript/README.md?plain=1#L421)
  2. [On line 518](https://github.com/reload/dpl-react/blob/docs-typescript/README.md?plain=1#L518)

- In the [Files section in code_guidelines.md](https://github.com/reload/dpl-react/blob/docs-typescript/docs/code_guidelines.md?plain=1#L94) there's being referred to `.js` and `.jsx` files but should that be changed to `.ts` and `.tsx`?

- It might be outside the scope of this PR but would it at a later point be worth considering how to define types if there's a need for defining types outside the scope of a component?

File: README.md [hunk 2]
```diff
@@ -343,130 +337,11 @@ __Cowabunga!__ You now got styling in your application
 ### Cross application components
 
 If the component is simple enough to be a primitive you would use in multiple
-occasions it's called an 'atom'. Such as a button or a link. If it's more
-specific that that and to be used across apps we just call it a component. An
-example would be some type of media presented alongside a header and some text.
+occasions it's called n component.
```
[the remark this answers is not in the document]
Comment (kasperg) [reply README.md]: It is a type. I have changed it to "a component".

File: README.md (hunk 1 above)
Comment (kasperg) [reply README.md]: It does. I have reverted this change in [e4f8186](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/78/commits/e4f81862eee59c08b6a037e9744dee68762e90ef).

**Comment by kasperg** [comment]
>  I have found a few places in the README where the term "atoms" is still present even though it seems as the usage of that term is discontinued

I agree about atom being discontinued in the project. I have removed references in [d1066ff](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/78/commits/d1066ff476410b3c0daf6bfcd566feb5a56768bc).

> there's being referred to .js and .jsx files but should that be changed to .ts and .tsx?

I agree. This has been changed in [4c1731f](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/78/commits/4c1731f1f8978be1d3221766ec0e56e593ab248a).

> [...] would it at a later point be worth considering how to define types if there's a need for defining types outside the scope of a component?

I agree. In current cross-project types are being added to `src/core`.

**Review by hrmoller — APPROVED** [review]
Look good to me😊

**Pull request merged**